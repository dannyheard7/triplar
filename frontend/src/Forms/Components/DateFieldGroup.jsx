import React from "react";
import Pikaday from 'pikaday'
import Moment from 'moment';

import "pikaday/css/pikaday.css";
import FormFieldGroup from "./FormFieldGroup";

export default class DateFieldGroup extends FormFieldGroup {

    constructor(props) {
        super(props);
        this.date_picker = React.createRef();
    }

    componentDidMount() {
        new Pikaday({
            field: this.date_picker.current,
            format: 'DD/MM/YYYY',
            onSelect: this.onChange
        })
    };

    formatValue = (value) => Moment(value).format('DD/MM/YYYY');

    render() {
        let value = this.formatValue(this.props.value);

        return (
            <div className="form-group">
                <label>{this.props.label}</label>
                <input name={this.props.name} type="text" autoComplete="off" value={value}
                       required={this.props.required} className="form-control" ref={this.date_picker}
                       data-parse="date" placeholder="DD/MM/YYYY" readOnly={true}/>
                {this.props.help && <small className="form-text text-muted">{this.props.help}</small>}
                {this.props.errors &&
                <li className="list-unstyled">
                    {this.props.errors.map(this.createErrorItem)}
                </li>
                }
            </div>
        );
    }
}