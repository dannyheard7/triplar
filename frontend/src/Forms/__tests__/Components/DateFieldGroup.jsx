import React from "react";
import {shallow} from "enzyme";
import DateFieldGroup from "../../Components/DateFieldGroup";
import ShallowRenderer from "react-test-renderer/shallow";

const renderer = new ShallowRenderer();
const faker = require('faker');

jest.mock('pikaday');

const pikaday = require('pikaday');

describe('<DateFieldGroup />', () => {

    test('renders correctly', () => {
        const result = renderer.render(<DateFieldGroup  value="2018-08-31"/>);
        expect(result).toMatchSnapshot();
    });

    test('Pikaday instance is created on mount', () => {
        const container = shallow(<DateFieldGroup name="date"/>);
        expect(pikaday).toBeCalledWith({"field": null, "format": "DD/MM/YYYY", "onSelect": undefined});
    });

    test('date is formatted correctly', () => {
        const container = shallow(<DateFieldGroup name="date"/>);
        let date = "2018-08-12";
        let formatted = container.instance().formatValue(date);
        expect(formatted).toEqual("12/08/2018");
    });

});