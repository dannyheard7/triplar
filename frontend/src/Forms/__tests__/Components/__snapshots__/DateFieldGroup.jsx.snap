// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DateFieldGroup /> renders correctly 1`] = `
<div
  className="form-group"
>
  <label />
  <input
    autoComplete="off"
    className="form-control"
    data-parse="date"
    name={undefined}
    placeholder="DD/MM/YYYY"
    required={undefined}
    type="text"
    value="31/08/2018"
  />
</div>
`;
